libpromises/evalfunction.c

```diff
@@ -238,8 +238,8 @@ static JsonElement* VarRefValueToJson(EvalContext *ctx, const FnCall *fp, const
             break;
 
         case RVAL_TYPE_CONTAINER:
-            convert = value;
-            *allocated = false;
+            convert = JsonCopy(value);
```

Comment:
OK, think about it: `convert` could be a new data element we've created from inline JSON or as a copy, **or** it could be a const reference (in which case `allocated` will be false).

In https://github.com/cfengine/core/pull/2514 I actually already fixed this warning by casting `value` to `JsonElement*`. I think that's the right fix.